File: gui/frontend/src/types/task.ts
/** Payload sent to POST /api/tasks to create a new task. */
export interface TaskCreate {
  name: string;
  description?: string;
  tags?: string[];
  template_id?: string;
  config_preset?: string;
  config_overrides?: Record<string, string>;
  env_vars?: Record<string, string>;
}

/** Task status returned by GET /api/tasks and GET /api/tasks/:id. */
export interface TaskStatus {
  id: string;
  name: string;
  status: "pending" | "running" | "completed" | "failed" | "stopped";
  template_id?: string;
  pid?: number;
  /** The resolved launch command that was or will be executed. */
  command?: string;
  /** ISO-8601 timestamp when the task was started. */
  started_at?: string;
  /** ISO-8601 timestamp when the task finished (if applicable). */
  finished_at?: string;
  /** Path to the log file on disk. */
  log_path?: string;
  /** Process exit code (if completed or failed). */
  exit_code?: number;
}

File: gui/backend/models/task.py
"""Pydantic models for task execution."""

from __future__ import annotations

from datetime import datetime
from enum import Enum

from pydantic import BaseModel, Field


class TaskStatusEnum(str, Enum):
    PENDING = "pending"
    RUNNING = "running"
    COMPLETED = "completed"
    FAILED = "failed"
    STOPPED = "stopped"


class TaskCreate(BaseModel):
    """Payload from the frontend task creation wizard."""

    name: str = ""
    description: str = ""
    tags: list[str] = []
    template_id: str | None = None
    config_preset: str | None = None
    config_overrides: dict[str, str] = {}
    env_vars: dict[str, str] = {}


class TaskStatus(BaseModel):
    id: str
    name: str = ""
    status: TaskStatusEnum
    template_id: str | None = None
    pid: int | None = None
    command: str = ""
    started_at: datetime = Field(default_factory=datetime.utcnow)
    finished_at: datetime | None = None
    log_path: str | None = None
    exit_code: int | None = None
